crates/typst-syntax/src/parser.rs

```diff
@@ -201,7 +205,7 @@ fn equation(p: &mut Parser) {
     let m = p.marker();
     p.enter_modes(LexMode::Math, AtNewline::Continue, |p| {
         p.assert(SyntaxKind::Dollar);
-        math(p, syntax_set!(Dollar, RightBracket, End));
+        math(p, syntax_set!(Dollar, End));
```

Comment:
This was a left over not-quite-bug from the parser refactor from a bit of careless find-replace. It's not actually a bug because we don't generate `RightBracket` tokens in math mode, but should definitely not be there.